test/core/iomgr/tcp_posix_test.cc

```diff
@@ -295,7 +298,7 @@ static void large_read_test(size_t slice_size) {
   GRPC_CLOSURE_INIT(&state.read_cb, read_cb, &state, grpc_schedule_on_exec_ctx);
 
   grpc_endpoint_read(ep, &state.incoming, &state.read_cb, /*urgent=*/false,
-                     /*min_progress_size=*/1);
+                     /*min_progress_size=*/min_progress_size);
```

Comment:
is there some kind of assert that we can drop in here to prove that min_progress_size is respected>

Reply:
I added an assert now. In the process, I discovered bugs in the way I was handling min_progress_size in tcp_posix.cc. It required some non-trivial amount of change (included in the latest commit). Could you ptal.